packages/mdc-textfield/foundation.js

```diff
@@ -365,11 +368,24 @@ class MDCTextFieldFoundation extends MDCFoundation {
     const {INVALID} = MDCTextFieldFoundation.cssClasses;
     if (isValid) {
       this.adapter_.removeClass(INVALID);
+      this.getNativeInput_().removeAttribute('aria-invalid');
     } else {
       this.adapter_.addClass(INVALID);
+      this.getNativeInput_().setAttribute('aria-invalid', true);
     }
+
     if (this.helperText_) {
       this.helperText_.setValidity(isValid);
+
+      if (isValid) {
```

Comment:
The nesting here can be reduced to improve legibility. I'd collapse the conditionals.

```
if (isValid && !this.initiallyHadDescribedBy_) {
} else if (!this.initiallyHadDescribedBy_) {}
```